save-plugins/fix-and-warn-plugin/src/commonMain/kotlin/org/cqfn/save/plugins/fixandwarn/FixAndWarnPlugin.kt

```diff
@@ -0,0 +1,161 @@
+package org.cqfn.save.plugins.fixandwarn
+
+import org.cqfn.save.core.config.TestConfig
+import org.cqfn.save.core.config.TestConfigSections
+import org.cqfn.save.core.files.readLines
+import org.cqfn.save.core.plugin.GeneralConfig
+import org.cqfn.save.core.plugin.Plugin
+import org.cqfn.save.core.result.TestResult
+import org.cqfn.save.plugin.warn.WarnPlugin
+import org.cqfn.save.plugins.fix.FixPlugin
+
+import okio.FileSystem
+import okio.Path
+
+private typealias WarningsList = MutableList<Pair<Int, String>>
+
+/**
+ * A plugin that runs an executable on a file, and combines two actions: fix and warn
+ * Plugin fixes test file, warns if something couldn't be auto-corrected after fix
+ * and compares output with expected output during one execution.
+ */
+class FixAndWarnPlugin(
+    testConfig: TestConfig,
+    testFiles: List<String>,
+    useInternalRedirections: Boolean = true) : Plugin(
+    testConfig,
+    testFiles,
+    useInternalRedirections) {
+    private val fixPluginConfig = testConfig.pluginConfigs.filterIsInstance<FixAndWarnPluginConfig>().single().fixPluginConfig
+    private val warnPluginConfig = testConfig.pluginConfigs.filterIsInstance<FixAndWarnPluginConfig>().single().warnPluginConfig
+    private val fixPlugin = FixPlugin(
+        createTestConfigForPlugins(TestConfigSections.FIX),
+        testFiles
+    )
+    private val warnPlugin = WarnPlugin(
+        createTestConfigForPlugins(TestConfigSections.WARN),
+        testFiles
+    )
+
+    /**
+     * Create TestConfig same as current, but with corresponding plugin configs list for nested [fix] and [warn] sections
+     *
+     * @param type type of nested section
+     * @return TestConfig for corresponding section
+     */
+    private fun createTestConfigForPlugins(type: TestConfigSections) = TestConfig(
+        testConfig.location,
+        testConfig.parentConfig,
+        mutableListOf(
+            testConfig.pluginConfigs.filterIsInstance<GeneralConfig>().single(),
+            if (type == TestConfigSections.FIX) {
+                fixPluginConfig
+            } else {
+                warnPluginConfig
+            }
+        )
+    )
+
+    override fun handleFiles(files: Sequence<List<Path>>): Sequence<TestResult> {
+        testConfig.validateAndSetDefaults()
+
+        val testFilePattern = warnPluginConfig.resourceNamePattern
+        val expectedFiles = files.filterTestResources(testFilePattern, match = false)
+
+        // Remove (in place) warnings from test files before fix plugin execution
+        val filesAndTheirWarningsMap = removeWarningsFromExpectedFiles(expectedFiles)
+
+        val fixTestResults = fixPlugin.handleFiles(files)
```

Comment:
This is connected with #133: instead of 
- reading the file and overwriting it to store warnings in memory
- reading it once more to compare with fixed
- restoring original contents for warn-plugin
we could support ignoring comments inside fix-plugin: it would read expected files for comparison and filter out warnings.